render_pdf/src/parser.py

```diff
@@ -0,0 +1,130 @@
+from lxml import etree
+from lxml.etree import fromstring
+import json
+from collections import defaultdict
+from urllib.request import urlopen
+
+def get_body(root):
+    return root.find(tagname("text")).find(tagname("body"))
+
+def get_title(root, alt_root):
+    # God this is ugly # TODO
+    try:
+        return root.find(tagname("head")).text
+    except:
+        try:
+            root = alt_root
+            tag = root.find(tagname("teiHeader"))\
+                    .find(tagname("fileDesc")) \
+                    .find(tagname("titleStmt"))
+            return tag.find(tagname("title")).text
+        except:
+            print("[Warning] Couldn't get title.")
+            return ""
+
+def get_author(root):
+    try:
+            tag = root.find(tagname("teiHeader"))\
+                    .find(tagname("fileDesc")) \
+                    .find(tagname("titleStmt"))
+            author = tag.find(tagname("author")).text
+            return author
+ 
+    except:
+        return ""
+
+def get_text_div(root):
+    return root.find(tagname("div"))
+
+def parse_lg(lg,verse=None):
+    phrases = defaultdict(dict)
+    if not verse:
+        verse = lg.get("{http://www.w3.org/XML/1998/namespace}id")
+    for line, l in enumerate(lg):
+        for seg in l:
```

Comment:
note that this assumes that all children in `<l>` are `<seg>` elements, which is not necessarily true. Might be better to find these explicitly.